[gdb/testsuite] Fix FAIL in gdb.base/annota1.exp

On openSUSE tumbleweed I run into:
...
FAIL: gdb.base/annota1.exp: run until main breakpoint (timeout)
...
due to a message related to libthread_db:
...
^Z^Zstarting^M
[Thread debugging using libthread_db enabled]^M
Using host libthread_db library "/lib64/libthread_db.so.1".^M
^M
^Z^Zframes-invalid^M
...
which is not matched by the regexp.

Fix this by updating the regexp.

Tested on x86_64-linux.

---
 gdb/testsuite/gdb.base/annota1.exp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gdb/testsuite/gdb.base/annota1.exp b/gdb/testsuite/gdb.base/annota1.exp
index 6e30c164ec9..046e942a321 100644
--- a/gdb/testsuite/gdb.base/annota1.exp
+++ b/gdb/testsuite/gdb.base/annota1.exp
@@ -138,6 +138,10 @@ set warning_gdb_index_re \
 	 "to use the section anyway\\."]
 set reading_re \
     "Reading \[^\r\n\]+"
+set libthread_db_re \
+    [multi_line \
+	 "\\\[Thread debugging using libthread_db enabled\\\]" \
+	 "Using host libthread_db library \[^\r\n\]+"]
 
 set run_re \
     [list \
@@ -145,7 +149,7 @@ set run_re \
 	 "\(\(\r\n$reading_re\)|\(\r\n$warning_slow_re\)\)*" \
 	 "\(\r\n$warning_gdb_index_re\)?" \
 	 "\(\(\r\n\r\n\032\032frames-invalid\)|\(\r\n\r\n\032\032breakpoints-invalid\)\)*\r\n\r\n" \
-	 "\032\032starting\(\(\r\n$reading_re\)|\(\r\n$warning_slow_re\)\)*" \
+	 "\032\032starting\(\(\r\n$reading_re\)|\(\r\n$warning_slow_re\)|\r\n$libthread_db_re\)*" \
 	 "\(\(\r\n\r\n\032\032frames-invalid\)|\(\r\n\r\n\032\032breakpoints-invalid\)\)*\r\n\r\n" \
 	 "\032\032breakpoint 1\r\n\r\n" \
 	 "Breakpoint 1, \r\n" \
